<launch>

  <node name="rplidarNode"          pkg="rplidar_ros"  type="rplidarNode" output="screen">
  <param name="serial_port"         type="string" value="/dev/ttyUSB_RPLIDAR"/>
  <param name="serial_baudrate"     type="int"    value="115200"/><!--A1/A2 -->
  <!--param name="serial_baudrate"     type="int"    value="256000"--><!--A3 -->
  <param name="frame_id"            type="string" value="laser"/>
  <param name="inverted"            type="bool"   value="false"/>
  <param name="angle_compensate"    type="bool"   value="true"/>
  </node>

  <arg name="config" default="$(find ca_driver)/config/default.yaml" />
  <arg name="desc" default="false" />

  <node name="ca_driver" pkg="ca_driver" type="ca_driver" output="screen">
    <rosparam command="load" file="$(arg config)" />
  </node>

  <node pkg="roomba_slam_auto_cpp" name="roomba" type="roomba" required="true">
    <param name="robot_model" value="CREATE_2" />
    <param name="loop_hz_origin" value="10" />
    <param name="boost" value="1" />
    <param name="x_hi" value="0.2" />
    <param name="x_lo" value="0.05" />
    <param name="x_start" value="0.01" />
    <param name="accel_step" value="0.01" />
    <param name="z_hi_origin" value="60.0" />
    <param name="z_lo_origin" value="30.0" />
    <param name="bumper_back_length" value="0.02" />
    <param name="cliff_back_length" value="0.05" />
    <param name="cliff_front_back_length" value="0.05" />
    <param name="bumper_turn_angle" value="40.0" />
    <param name="cliff_turn_angle" value="70.0" />
    <param name="cliff_front_turn_angle" value="160.0" />
    <param name="bumper_signal_threshold" value="200" />
    <param name="bumper_signal_front_threshold" value="150" />
    <param name="bumper_signal_center_threshold" value="100" />
  </node>
</launch>
